lib/igniter/test.ex

```diff
@@ -344,128 +393,226 @@ defmodule Igniter.Test do
+      /tmp/
+      """,
+      assigns
+    )
+    |> put_file(
+      "mix.exs",
+      """
+      defmodule <%= @mod %>.MixProject do
+        use Mix.Project
+
+        def project do
+          [
+            app: :<%= @app %>,
+            version: "0.1.0",
+            elixir: "~> <%= @version %>",
+            start_permanent: Mix.env() == :prod,
+            deps: deps()
+          ]
+        end
 
-    # If you run "mix test --cover", coverage assets end up here.
-    /cover/
+        # Run "mix help compile.app" to learn about applications.
+        def application do
+          [
+            extra_applications: [:logger]<%= @sup_app %>
+          ]
+        end
 
-    # The directory Mix downloads your dependencies sources to.
-    /deps/
+        # Run "mix help deps" to learn about dependencies.
+        defp deps do
+          [
+            # {:dep_from_hexpm, "~> 0.3.0"},
+            # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"}
+          ]
+        end
+      end
+      """,
+      assigns
+    )
+    |> put_file(
+      "lib/#{mod_filename}.ex",
+      """
+      defmodule <%= @mod %> do
+        @moduledoc \"""
+        Documentation for `<%= @mod %>`.
+        \"""
 
-    # Where third-party dependencies like ExDoc output generated docs.
-    /doc/
+        @doc \"""
+        Hello world.
 
-    # Ignore .fetch files in case you like to edit your project deps locally.
-    /.fetch
+        ## Examples
 
-    # If the VM crashes, it generates a dump, let's ignore it too.
-    erl_crash.dump
+            iex> <%= @mod %>.hello()
+            :world
 
-    # Also ignore archive artifacts (built via "mix archive.build").
-    *.ez
+        \"""
+        def hello do
+          :world
+        end
+      end
+      """,
+      assigns
+    )
+    |> put_file(
+      "test/test_helper.exs",
+      """
+      ExUnit.start()
+      """,
+      assigns
+    )
+    |> put_file(
+      "test/#{mod_filename}_test.exs",
+      """
+      defmodule <%= @mod %>Test do
+        use ExUnit.Case
+        doctest <%= @mod %>
 
-    # Ignore package tarball (built via "mix hex.build").
-    #{app_name}-*.tar
+        test "greets the world" do
+          assert <%= @mod %>.hello() == :world
+        end
+      end
+      """,
+      assigns
+    )
+  end
 
-    # Temporary files, for example, from tests.
-    /tmp/
-    """)
-    |> Map.put_new("mix.exs", """
-    defmodule #{module_name}.MixProject do
-      use Mix.Project
+  # https://github.com/phoenixframework/phoenix/tree/ed1331ea71bf458cc7909bca023564905a44fa90/installer
+  # TODO: opts[:umbrella]
+  # TODO: maybe support other options such as --module, --no-ecto, etc
+  defp add_phoenix_new(opts) do
+    # FIXME: base path
+    app_name = to_string(opts[:app_name] || :test)
+    source_base_path = Path.expand("../../deps/phx_new/templates", __DIR__)
+
+    project =
+      app_name
+      |> Phx.New.Project.new([])
```

Comment:
We can't use this like this if we have only a dev/test dependency on `phx_new`. It would cause compilation errors/warnings in any application that wants to use it. So to do this we'd need to make `phx_new` an optional *non* dev/test dependency of igniter, and then do something like this:

```elixir
if Code.ensure_loaded?(Phx.New) do
  def phoenix_project(opts \\ []) do
    ...current implementation
  end
else
  def phoenix_project(opts \\ []) do
     raise "You must include the `phx_new` dependency to generate to use `#{inspect(__MODULE__)}.phoenix_project/1`"
  end
end
```